kf/nfconfig.go

```diff
@@ -1018,73 +1070,58 @@ func (c *NFConfigs) AddProgramWithoutChaining(ifaceName string, bpfProgs *models
-				}
-			} else {
-				prog := c.IngressXDPBpfs[ifaceName].Front().Value.(*BPF)
-				return fmt.Errorf("failed to add %v due to existing program %v on iface %v direction %v", bpfProg.Name, prog.Program.Name, ifaceName, models.XDPIngressType)
-			}
-		}
+	if err := c.addProgramByHookWithoutChaining(ifaceName, models.XDPIngressType, c.IngressXDPBpfs, bpfProgs.XDPIngress); err != nil {
+		return err
 	}
-
-	if len(bpfProgs.TCIngress) == 1 {
-		bpfProg := bpfProgs.TCIngress[0]
-		if bpfProg.AdminStatus == models.Enabled {
-			if c.IngressTCBpfs[ifaceName] == nil {
-				c.IngressTCBpfs[ifaceName] = list.New()
-				if err := c.PushBackAndStartBPF(bpfProg, ifaceName, models.IngressType); err != nil {
-					return fmt.Errorf("failed to PushBackAndStartBPF BPF Program: %v", err)
-				}
-			} else {
-				prog := c.IngressTCBpfs[ifaceName].Front().Value.(*BPF)
-				return fmt.Errorf("failed to add %v due to existing program %v on iface %v direction %v", bpfProg.Name, prog.Program.Name, ifaceName, models.IngressType)
-			}
-		}
+	if err := c.addProgramByHookWithoutChaining(ifaceName, models.IngressType, c.IngressTCBpfs, bpfProgs.TCIngress); err != nil {
+		return err
 	}
+	if err := c.addProgramByHookWithoutChaining(ifaceName, models.EgressType, c.EgressTCBpfs, bpfProgs.TCEgress); err != nil {
+		return err
+	}
+
+	return nil
+}
 
-	if len(bpfProgs.TCEgress) == 1 {
-		bpfProg := bpfProgs.TCEgress[0]
+// addProgramByHookWithoutChaining
+func (c *NFConfigs) addProgramByHookWithoutChaining(ifaceName, progType string, progMap map[string]*list.List, bpfProgs []*models.BPFProgram) error {
+	if len(bpfProgs) == 1 {
+		bpfProg := bpfProgs[0]
 		if bpfProg.AdminStatus == models.Enabled {
-			if c.EgressTCBpfs[ifaceName] == nil {
-				c.EgressTCBpfs[ifaceName] = list.New()
-				if err := c.PushBackAndStartBPF(bpfProg, ifaceName, models.EgressType); err != nil {
-					return fmt.Errorf("failed to PushBackAndStartBPF BPF Program: %v", err)
+			if progMap[ifaceName] == nil {
+				progMap[ifaceName] = list.New()
+				if err := c.PushBackAndStartBPF(bpfProg, ifaceName, progType); err != nil {
+					return fmt.Errorf("failed to PushBackAndStartBPF BPF Program: %w", err)
 				}
 			} else {
-				prog := c.EgressTCBpfs[ifaceName].Front().Value.(*BPF)
-				return fmt.Errorf("failed to add %v due to existing program %v on iface %v direction %v", bpfProg.Name, prog.Program.Name, ifaceName, models.EgressType)
+				prog := progMap[ifaceName].Front().Value
+				return fmt.Errorf("failed to add %v due to existing program %v on iface %v direction %v", bpfProg.Name, prog.Name(), ifaceName, progType)
 			}
 		}
 	}
 	return nil
 }
 
 // AddProgramsOnInterface: AddProgramsOnInterface will add given ebpf programs on given interface
-func (c *NFConfigs) AddProgramsOnInterface(ifaceName, HostName string, bpfProgs *models.BPFPrograms) error {
+func (c *NFConfigs) addProgramsOnInterface(ifaceName, HostName string, bpfProgs *models.BPFPrograms) error {
```

Comment:
This changes the capitalization to no longer match the comment in 1108

Reply:
My comment is mainly about the inconsistency between the comment and the function name.  I have no opinion on whether it should be capitalized or not, only that the two should be consistent, which they're currently not in this PR.

Reply:
Sorry, I missed:
>  match the comment in 1108
Thank you for pointing this out.